apps/src/JSInterpreter.js

```diff
@@ -649,9 +653,6 @@ JSInterpreter.prototype.selectCurrentCode = function (highlightClass) {
  * of the userCode area, the return value is -1
  */
 JSInterpreter.prototype.getUserCodeLine = function () {
-  if (this.studioApp.hideSource) {
-    return -1;
-  }
```

Comment:
My concern with deleting this is that we've made the shared apps (`hideSource === true`) slower. The good news is they probably should have been running slower for cases where we weren't running at maxSpeed. But I wonder if you can provide a fast exit for the `maxSpeed` case? Things running too slowly in the interpreter has been a recurring theme, and keeping the "shared app" case running at the fastest possible speed seems like a good thing.

For your test cases below, you can probably ensure that you're not running at maxSpeed, true?